docs: rfc-0241 tari-monero atomic swap (#3623)
tari-project/tari: RFC/src/RFC-0202_TariScriptOpcodes.md

Identical to [`CheckSig`](#checksigmsg), except that nothing is pushed to the stack if the signature is valid, and the
operation fails with `VERIFY_FAILED` if the signature is invalid.

> ##### Ristretto,

Reviewer: Ristretto is a bit vague should be clear we are calculating a pubkey from a secret key. Maybe something like RistrettoKeyCalc?

> Pop the top element, converts it to a public_key, and pushes this to the stack.

Reviewer: ```suggestion
Pop the top element, converts it to a Ristretto public_key, and pushes this to the stack.
```

Identical to [`CheckSig`](#checksigmsg), except that nothing is pushed to the stack if the signature is valid, and the
operation fails with `VERIFY_FAILED` if the signature is invalid.

> ##### RistrettoPoint,

Reviewer 2: ```suggestion
##### ConvertToRistrettoPoint,
```

Reviewer 2: What format is the data that's on the stack? Pure bytes?

> ##### ConvertToRistrettoPoint,

Reviewer 3: ```suggestion
##### ToRistrettoPoint,
```
As descriptive with fewer keystrokes. ditto for other opcodes

> Pop the top element, converts it to a Ristretto point, and pushes this to the stack.

Reviewer 4: It's not clear what the source element would be for this conversion. Of the stack items allowed (Number,  Hash,    Commitment,    PublicKey,    Signature) non really fit as inputs to this op, are you proposing a 32-byte blob or would this be a conversion from `PublicKey`? `Hash` is 32 bytes but I would say it isn't appropriate for this use.

Author: Yeah, I realized this, and for now, the script mentions Hash as it can be used as 32 bytes. Left his out for now till we can discuss what route we want forward with this.

Reviewer 3: The correct approach in my view would be to define a new `StackItem` type `RistrettoScalar(RisrettoSecretKey)`

```suggestion
Posp the top element, which must be a valid Ristretto scalar, calculates the corresponding Ristretto point and pushes this to the stack.
```

Reviewer 4: Ah yup that makes sense

* Fails with `EMPTY_STACK` if the stack is empty.
* Fails with `INVALID_INPUT` if the top stack element is not a input.

> ##### Ed25519Point,

Reviewer 2: ```suggestion
##### ConvertToEd25519Point,
```

Pop the top element, converts it to a Ed25519 point, and pushes this to the stack.

* Fails with `EMPTY_STACK` if the stack is empty.
* Fails with `INVALID_INPUT` if the top stack element is not a input.

> ##### Secp256k1Point,

Reviewer 2: ```suggestion
##### ConvertToSecp256k1Point,
```
